openedx/adg/lms/applications/tests/test_admin/test_inlines.py

```diff
@@ -61,3 +70,132 @@ def test_responsibilities_work_experience_inline(work_experience):
+    }
+
+    CourseInlineFormset = inlineformset_factory(
+        MultilingualCourseGroup,
+        MultilingualCourse,
+        formset=MultilingualCourseInlineFormset,
+        fields=['course', 'multilingual_course_group']
+    )
+    formset = CourseInlineFormset(data, prefix='multilingual_courses')
+    assert formset.is_valid()
+
+
+@pytest.mark.django_db
+@mock.patch('openedx.adg.lms.applications.admin.modulestore')
+def test_multilingual_group_two_courses_of_same_language(mock_module_store):
+    """
+    Test formset is not valid for multiple courses with same language.
+    """
+    mock_module_store.get_course.return_value = mock.Mock()
+    mock_module_store.get_course.language.return_value = 'en'
+    course1 = CourseOverviewFactory()
+    course2 = CourseOverviewFactory()
+    data = {
+        'multilingual_courses-TOTAL_FORMS': '2',
+        'multilingual_courses-INITIAL_FORMS': '0',
+        'multilingual_courses-0-id': '',
+        'multilingual_courses-0-multilingual_course_group': '',
+        'multilingual_courses-0-course': course1.id,
+        'multilingual_courses-1-id': '',
+        'multilingual_courses-1-multilingual_course_group': '',
+        'multilingual_courses-1-course': course2.id,
+        'name': 'Test',
+        'is_prerequisite': 'on',
+    }
+
+    CourseInlineFormset = inlineformset_factory(
+        MultilingualCourseGroup,
+        MultilingualCourse,
+        formset=MultilingualCourseInlineFormset,
+        fields=['course', 'multilingual_course_group']
+    )
+    formset = CourseInlineFormset(data, prefix='multilingual_courses')
+    assert not formset.is_valid()
+
+
+@pytest.mark.django_db
+@mock.patch('openedx.adg.lms.applications.admin.modulestore')
+def test_multilingual_group_course_already_exists(mock_module_store):
+    """
+    Test formset is not valid for an existing multilingual versioned course.
+    """
+    mock_module_store.get_course.return_value = mock.Mock()
+    course1 = MultilingualCourseFactory()
+    data = {
+        'multilingual_courses-TOTAL_FORMS': '2',
+        'multilingual_courses-INITIAL_FORMS': '0',
+        'multilingual_courses-0-id': '',
+        'multilingual_courses-0-multilingual_course_group': '',
+        'multilingual_courses-0-course': course1.course.id,
+        'multilingual_courses-1-id': '',
+        'multilingual_courses-1-multilingual_course_group': '',
+        'multilingual_courses-1-course': '',
+        'name': 'Test',
+        'is_prerequisite': 'on',
+    }
+
+    CourseInlineFormset = inlineformset_factory(
+        MultilingualCourseGroup,
+        MultilingualCourse,
+        formset=MultilingualCourseInlineFormset,
+        fields=['course', 'multilingual_course_group']
+    )
+    formset = CourseInlineFormset(data, prefix='multilingual_courses')
+    assert not formset.is_valid()
+
+    error = formset.errors[0]['course']
+    assert error == ['Multilingual course with this Multilingual version of a course already exists.']
```

Comment:
We can remove all this repetition in these tests by [parametrizing](https://docs.pytest.org/en/stable/parametrize.html) them.